WalletWasabi/WabiSabi/Backend/Rounds/CoinJoinFeeRateStatistics.cs

```diff
@@ -0,0 +1,52 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.WabiSabi.Backend.Rounds.CoinJoinStorage;
+
+namespace WalletWasabi.WabiSabi.Backend.Rounds;
+
+public class CoinJoinFeeRateStatistics
+{
+	private const int MaximumDaysToStore = 30;
+	private List<DateTimeAndFeeRate> FeeRates { get; }
+
+	private CoinJoinFeeRateStatistics(IEnumerable<DateTimeAndFeeRate> dateAndfeeRates)
+	{
+		FeeRates = new(dateAndfeeRates.OrderBy(x => x.DateTimeOffset));
+	}
+
+	public static CoinJoinFeeRateStatistics LoadFromCoinJoinTransactionArchiver(CoinJoinTransactionArchiver coinJoinTransactionArchiver)
+	{
+		DateTimeOffset now = DateTimeOffset.UtcNow;
+		var from = now - TimeSpan.FromDays(MaximumDaysToStore);
+		var dateAndFeeRates = coinJoinTransactionArchiver.ReadJson(from, now)
+			.Select(txinfo => new DateTimeAndFeeRate(
+				DateTimeOffset.FromUnixTimeMilliseconds(txinfo.Created),
+				new FeeRate(txinfo.FeeRate)));
+
+		return new(dateAndFeeRates);
+	}
+
+	public void Add(FeeRate feeRate)
+	{
+		DateTimeOffset now = DateTimeOffset.UtcNow;
+
+		FeeRates.Add(new(now, feeRate));
+
+		// Prune old items.
+		DateTimeOffset removeBefore = now - TimeSpan.FromDays(MaximumDaysToStore);
+		while (FeeRates.Any() && (FeeRates[0].DateTimeOffset < removeBefore))
+		{
+			FeeRates.RemoveAt(0);
+		}
+	}
+
+	public FeeRate GetAvarage(DateTimeOffset from)
+	{
+		return new FeeRate(FeeRates.Where(x => x.DateTimeOffset >= from).Average(x => x.FeeRate.SatoshiPerByte));
+	}
```

Comment:
I'm not sure what time period is considered for the average really but an alternative idea might be "weighted average" where "more recent fee rates has higher impact".

The most simple approach would be to assign weights 1, 2,3,4,.., n to `FeeRates`. Not ideal either, but probably good. 

It would be certainly useful to have a chart of fee rates for some longer time period that can validate the idea.